src/vecs/collection.py

```diff
@@ -340,33 +340,46 @@ def fetch(self, ids: Iterable[str]) -> List[Record]:
                     records.extend(chunk_records)
         return records
 
-    def delete(self, ids: Iterable[str]) -> List[str]:
+    def delete(self, ids: Optional[Iterable[str]] = None, filters: Optional[Metadata] = None) -> List[str]:
         """
-        Deletes vectors from the collection by their identifiers.
+        Deletes vectors from the collection by matching filters or ids.
 
         Args:
             ids (Iterable[str]): An iterable of vector identifiers.
+            filters (Metadata): A dictionary of metadata key-value pairs to match.
 
         Returns:
             List[str]: A list of the identifiers of the deleted vectors.
         """
-        if isinstance(ids, str):
-            raise ArgError("ids must be a list of strings")
+        if ids is None and filters is None:
+            raise VectorDeletionError("Either ids or metadata must be provided.")
 
-        chunk_size = 12
+        del_ids = []
 
-        del_ids = list(ids)
-        ids = []
         with self.client.Session() as sess:
             with sess.begin():
-                for id_chunk in flu(del_ids).chunk(chunk_size):
+                if ids is not None:
```

Comment:
lets move all client side checks outside the database transaction to keep them as fast as possible.
coalescing `None` to an empty list would probably work best for keeping the logic in the db transaction clear